src/v/s3/client.cc

```diff
@@ -304,7 +304,7 @@ request_creator::make_list_objects_v2_request(
         header.insert(aws_header_names::start_after, std::to_string(*max_keys));
     }
     auto ec = _sign.sign_header(header, aws_signatures::emptysig);
-    vlog(s3_log.trace, "ListObjectsV2:\n {}", header);
+    vlog(s3_log.trace, "ListObjectsV2:\n {}", http::redacted_header(header));
```

Comment:
remembering how to wrap the header every time we log it is hard to enforce. can we use the type system to enforce this? one option is to make the default operator<< overload redacted, and then provide a separate non-redacted accessor.

Reply:
thanks, @LenaAn had the same suggestion to overload `operator<<` - I will explore overloading in a new PR.

Reply:
created #5011 for this issue

Reply:
Raised https://github.com/redpanda-data/redpanda/pull/6067 for this - it was tricky to overload the operator without subclassing (I actually couldn't see how to do it). I implemented a custom fmt::formatter instead.